src/form.ts

```diff
@@ -409,12 +409,22 @@ function useForm<T extends object, C>(
             const stateErrored = analyzeErrors(stateErrors);
             if (stateErrored) {
                 const value = accumulateOnError
-                    ? accumulateValues(state.value, schema, { nullable: true })
-                    : undefined;
+                    ? accumulateValues(
+                        state.value,
+                        schema,
+                        { nullable: true },
+                        state.value, context,
```

Comment:
We were thinking about passing the immediate parent value, right?

Also `state.value, context,` should be on separate lines